confluent_kafka/src/Producer.c

```diff
@@ -252,7 +252,7 @@ int32_t Producer_partitioner_cb (const rd_kafka_topic_t *rkt,
 }
 
 
-#if HAVE_PRODUCEV
+#if RD_KAFKA_V_HEADERS
```

Comment:
I would probably stick this just around the headers argument in the function argument list, to avoid duplication.